src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -23551,23 +23551,19 @@ public void IsOperator_07()
 class Helper1<T, U>
     where T : allows ref struct
```

Comment:
Are we testing the following where `U` has `allows ref struct`?
```csharp
class Helper1<T, U>
{
    public static void Test1(U h1)
        where U : allows ref struct
    {
        if (h1 is T)
        ...
    }
}

Helper1<I1, S>.Test1(new S());
...
```

Reply:
The previous test? The scenario that you suggest (T  has no constraint) is not affected by the change. Doesn't take the affected code path:
```
if (targetType is TypeParameterSymbol { AllowsRefLikeType: true })
```

Reply:
> The previous test?

It's not clear to me why `is` check in the previous test is a compile-time error but the `is` check in this test is a runtime check.

> The scenario that you suggest (T  has no constraint) is not affected by the change.

That's true, thanks.

Reply:
> It's not clear to me why `is` check in the previous test is a compile-time error but the `is` check in this test is a runtime check.

The runtime check for this test is safe because we know that the value being checked is never a ref struct. That is not true for the previous test

Reply:
We can box the value being checked in this test

Reply:
> The scenario that you suggest (T has no constraint) is not affected by the change.

Consider testing if it's not already tested, even though it is not affected by this change.

Reply:
I think it is covered by IsOperator_05